### PR TITLE
Remove vendoring

### PR DESCRIPTION
**Pull request by MartinWeindel**
**What this PR does / why we need it**:
Remove vendoring and update dependencies

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other developer
Remove vendoring
```

```other operator
Bump kubernetes dependencies from `v0.26.4` to `v0.29.0`.
Bump github.com/aws/aws-sdk-go from `v1.42.25` to `v1.49.11`.
```

```other operator
Bump golang from `1.20.4` to `1.21.5`
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -5,8 +5,11 @@ package main
 
 import (
 	"fmt"
+
 	"github.com/gardener/aws-custom-route-controller/pkg/util/logger"
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+
```
Comment (ScheererJ) [inline main.go:12]: The `import`s look kind of strange. Would we not want the common standard library components, i.e. `fmt`, `os` and `time`, be grouped together with the other libraries and the current project being separated?

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Thanks for the nice improvement. I just have one minor remark.

File: main.go (hunk 1 above)
Comment (MartinWeindel) [reply main.go]: added golangci-lint to `check` make target.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**